server/src/main/java/org/apache/druid/client/JsonParserIterator.java

```diff
@@ -185,6 +185,12 @@ private void init()
         } else if (nextToken == JsonToken.START_OBJECT) {
           throw convertException(jp.getCodec().readValue(jp, QueryException.class));
         } else {
+          if (nextToken == JsonToken.VALUE_STRING) {
+            LOG.error(
+                "Next Token value is of type VALUE_STRING with content [%s], not as expected START_ARRAY",
+                jp.getText()
+            );
+          }
           throw convertException(
               new IAE("Next token wasn't a START_ARRAY, was[%s] from url[%s]", jp.getCurrentToken(), url)
```

Comment:
Should we add `jp.getText()` in the exception. That way its a much cleaner implementation?
We should chom jp.getText() to around 200 chars before adding it to the message though.

Reply:
added a unit test and chop the error message at most to 192.